.github/workflows/mypy.yaml

```diff
@@ -19,15 +19,12 @@ jobs:
       - name: Set up python
```

Comment:
There is a reusable github action for mypy that will pip install the requirements.txt and types.txt files. We should consider using that. (it doesn't know about things like `[dev,typing]` at the moment but in theory could be changed to provide an alternative mechanism over the .txt files.

Reply:
The "most correct" path from here would be to migrate the `setup.cfg` to `pyproject.toml` and use `[dependency-groups]` together with a `uv` to perform all tasks.

I made these particular CI changes to align the "source of truth" about what packages are needed to perform what tasks with the project's dependency graph instead of having bespoke extra pip-install tasks in the workflows. Packaging-wise, it doesn't make sense for these targets to be "extras" in the sense that that has a specific meaning.

Reply:
We should consider tweaking the reusable mypy workflow to do it your way, although it would need a parameter to tell it whether it's doing it with dependency groups or requirements.txt.

Reply:
This one? https://github.com/lsst/rubin_workflows

I can take a closer look at it.

Reply:
Yes (I edited my earlier message to mention it but easy to miss).